<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Python modules for OpenStructure &mdash; Python modules for OpenStructure 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/sphinxdoc.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="Python modules for OpenStructure 1.0 documentation" href="#" />
    <link rel="next" title="align_traj_on_density module" href="align_traj_on_density.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="align_traj_on_density.html" title="align_traj_on_density module"
             accesskey="N">next</a> |</li>
        <li><a href="#">Python modules for OpenStructure 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="#">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Python modules for OpenStructure</a><ul>
<li><a class="reference internal" href="#introduction">Introduction</a></li>
<li><a class="reference internal" href="#installation">Installation</a></li>
<li><a class="reference internal" href="#available-modules">Available modules</a></li>
<li><a class="reference internal" href="#cite-us">Cite Us</a></li>
</ul>
</li>
<li><a class="reference internal" href="#indices-and-tables">Indices and tables</a></li>
</ul>

  <h4>Next topic</h4>
  <p class="topless"><a href="align_traj_on_density.html"
                        title="next chapter">align_traj_on_density module</a></p>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/index.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="python-modules-for-openstructure">
<h1>Python modules for OpenStructure<a class="headerlink" href="#python-modules-for-openstructure" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
</div>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>This is a set of python modules based on the <a class="reference external" href="http://www.openstructure.org">OpenStructure</a> framework and mainly aimed at performing various analyses of molecular dynamics simulations but also offers some functions for sequences and molecular structures.</p>
<p>The functions are roughly clustered into modules, which are rather strongly interdependent. These modules provide functions for classical analyses of MD trajectories such as principal components analysis, correlations (covariance, mutual information), helper functions to make nice movies, but also more advanced  analysis tools notably for simulations of lipidic systems. For example there are functions to <a class="reference internal" href="align_traj_on_density.html"><em>align frames</em></a> from a simulation of complex lipidic assemblies and calculate <a class="reference internal" href="surface_alg.html"><em>the local curvatures</em></a> of the membrane (see ref. <a class="footnote-reference" href="#id17" id="id1">[2]</a> and <a class="footnote-reference" href="#id18" id="id2">[3]</a>), or to extract <a class="reference internal" href="lipid_analysis.html"><em>the elastic moduli</em></a> of the membrane from the simulation (ref. <a class="footnote-reference" href="#id19" id="id3">[4]</a>, <a class="footnote-reference" href="#id20" id="id4">[5]</a> and <a class="footnote-reference" href="#id21" id="id5">[6]</a>).</p>
<p>Several modules from this set have already been moved into <em>OpenStructure</em>, as for example for calculating alpha helical kink angles (<a class="reference external" href="http://www.openstructure.org/docs/1.4/mol/alg/molalg/">mol.alg.helix_kinks</a>), hbonds (mol.alg.hbond) or basic functions for structure and trajectory analysis (<a class="reference external" href="http://www.openstructure.org/docs/1.4/mol/alg/molalg/">mol.alg.structure_analysis</a> and <a class="reference external" href="http://www.openstructure.org/docs/1.4/mol/alg/molalg/">mol.alg.trajectory_analysis</a>). Some of the modules or functions in the modules presented here might also migrate into <em>OpenStructure</em> in the future.</p>
</div>
<div class="section" id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h2>
<p>Be aware that these modules depend on <a class="reference external" href="http://www.openstructure.org">OpenStructure</a> and on several standard scientific python packages.
Help on installation and basic usage can be found <a class="reference internal" href="installation.html"><em>here</em></a>.</p>
</div>
<div class="section" id="available-modules">
<h2>Available modules<a class="headerlink" href="#available-modules" title="Permalink to this headline">¶</a></h2>
<ul>
<li><dl class="first docutils">
<dt>Several modules are aimed at analyzing lipidic systems.</dt>
<dd><ul class="first last simple">
<li><a class="reference internal" href="align_traj_on_density.html"><em>Aligning trajectories</em></a> without mtaching atoms of different frames. This is done using densities instead and is typically used to align membranes or complex lipidic systems.</li>
<li>Calculating <a class="reference internal" href="lipid_analysis.html"><em>elastic properties</em></a> of membranes.</li>
<li>Calculating <a class="reference internal" href="surface_alg.html"><em>normals and curvatures</em></a> on surfaces.</li>
<li>A module to calculate <a class="reference internal" href="lipid_order_parameters.html"><em>lipid order parameters</em></a></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>Modules aimed at analyzing trajectories of proteins.</dt>
<dd><ul class="first last simple">
<li><a class="reference internal" href="principal_components.html"><em>Principal components analysis</em></a></li>
<li>Other functions to <a class="reference internal" href="trajectory_utilities.html"><em>analyze trajectories</em></a> such as covariance, mutual information, correlations and so on.</li>
<li>Calculate and plot <a class="reference internal" href="secondary_structure.html"><em>protein secondary structure</em></a>, both for structures and trajectories. This module depends on the binding of openstructure to dssp.</li>
<li>Calculate the <tt class="xref doc docutils literal"><span class="pre">size</span> <span class="pre">of</span> <span class="pre">a</span> <span class="pre">pore</span></tt> in a channel or other protein.</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>Graphical output:</dt>
<dd><ul class="first last simple">
<li>Helper functions for <a class="reference internal" href="plot.html"><em>plotting</em></a></li>
<li>Helper functions to make nice <a class="reference internal" href="movies.html"><em>movies</em></a></li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Some functions to work on <a class="reference internal" href="sequence_alg.html"><em>sequences and alignments</em></a> (e.g. search for motifs)</p>
</li>
<li><dl class="first docutils">
<dt>Miscellaneous:</dt>
<dd><ul class="first last simple">
<li>Helper functions to <a class="reference internal" href="file_utilities.html"><em>read/write files</em></a> and parse some information from PDB files.</li>
<li>Some functions to perform  <a class="reference internal" href="clustering.html"><em>clustering</em></a>.</li>
<li>Correctly build bonds for <a class="reference internal" href="coarse_grained_conop.html"><em>Martini coarse-grained systems</em></a>.</li>
<li>Some helper functions to work with <tt class="xref doc docutils literal"><span class="pre">periodic</span> <span class="pre">boundary</span> <span class="pre">conditions</span></tt>.</li>
<li>Some support for reading <a class="reference internal" href="colvar.html"><em>NAMD COLVAR</em></a> files.</li>
</ul>
</dd>
</dl>
</li>
</ul>
</div>
<div class="section" id="cite-us">
<h2>Cite Us<a class="headerlink" href="#cite-us" title="Permalink to this headline">¶</a></h2>
<p>If you use any of this code, please cite ref. <a class="footnote-reference" href="#id16" id="id7">[1]</a> for the use of OpenStructure.
Moreover if you use this code to align trajectories using the densities, please cite ref. <a class="footnote-reference" href="#id17" id="id8">[2]</a>, for calculations of curvature please cite ref. <a class="footnote-reference" href="#id18" id="id9">[3]</a> and for calulations of lipid elastic properties please cite ref. <a class="footnote-reference" href="#id19" id="id10">[4]</a>. The theory behind the calculation of elastic properties and some applications can be found in refs. <a class="footnote-reference" href="#id20" id="id11">[5]</a> and <a class="footnote-reference" href="#id21" id="id12">[6]</a>.</p>
<p>Moreover some of these modules also depend on other python libraries, notably
<a class="reference external" href="http://www.numpy.org">numpy</a>, <a class="reference external" href="http://www.scipy.org">scipy</a> and <a class="reference external" href="http://matplotlib.org">matplotlib</a>. If you use any function from this library in your work, please cite numpy (for example ref. <a class="footnote-reference" href="#id22" id="id13">[7]</a>). If you use any plots generated by these modules please cite matplotlib (ref. <a class="footnote-reference" href="#id23" id="id14">[8]</a>). Finally if you use these modules to align trajectories, calculate membrane elastic properties or for clustering please cite scipy (ref. <a class="footnote-reference" href="#id24" id="id15">[9]</a>). Other appropriate citations for these tools can be found on the <a class="reference external" href="http://www.scipy.org/citing.html">scipy website</a></p>
<table class="docutils footnote" frame="void" id="id16" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id7">[1]</a></td><td>Marco Biasini, T. Schmidt, S. Bienert, V. Mariani, G. Studer, J. Haas, N. Johner, A.D. Schenk, A. Philippsen and T. Schwede, &#8220;OpenStructure: an integrated software framework for computational structural biology&#8221;, Acta Cryst., 2013</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="id17" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[2]</td><td><em>(<a class="fn-backref" href="#id1">1</a>, <a class="fn-backref" href="#id8">2</a>)</em> George Khelashvili, P. Blecua Carrillo Albornoz, N. Johner, S. Mondal, M. Caffrey, and H. Weinstein. &#8220;Why GPCRs Behave Differently in Cubic and Lamellar Lipidic Mesophases.&#8221;, Journal of the American Chemical Society 134, no. 38 (September 26, 2012): 15858–68.</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="id18" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[3]</td><td><em>(<a class="fn-backref" href="#id2">1</a>, <a class="fn-backref" href="#id9">2</a>)</em> Niklaus Johner, S. Mondal, G. Morra, M. Caffrey, H. Weinstein, and G. Khelashvili. &#8220;Protein and Lipid Interactions Driving Molecular Mechanisms of in Meso Crystallization.&#8221; Journal of the American Chemical Society 136, no. 8 (February 26, 2014): 3271–84.</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="id19" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[4]</td><td><em>(<a class="fn-backref" href="#id3">1</a>, <a class="fn-backref" href="#id10">2</a>)</em> Niklaus Johner, D. Harries and G. Khelashvili, &#8220;Implementation of a methodology for determining elastic properties of lipid assemblies from molecular dynamics simulations&#8221;, submitted to BMC Bioinformatics (2015)</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="id20" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[5]</td><td><em>(<a class="fn-backref" href="#id4">1</a>, <a class="fn-backref" href="#id11">2</a>)</em> George Khelashvili, N. Johner, G. Zhao, D. Harries, and H. L. Scott. &#8220;Molecular Origins of Bending Rigidity in Lipids with Isolated and Conjugated Double Bonds: The Effect of Cholesterol.&#8221; Chemistry and Physics of Lipids 178 (February 2014): 18–26.</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="id21" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[6]</td><td><em>(<a class="fn-backref" href="#id5">1</a>, <a class="fn-backref" href="#id12">2</a>)</em> Niklaus Johner, D. Harries, and G. Khelashvili. &#8220;Curvature and Lipid Packing Modulate the Elastic Properties of Lipid Assemblies: Comparing HII and Lamellar Phases.&#8221; The Journal of Physical Chemistry Letters 5, no. 23 (December 4, 2014): 4201–6.</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="id22" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id13">[7]</a></td><td>Stéfan van der Walt, S. C. Colbert and G. Varoquaux. &#8220;The NumPy Array: A Structure for Efficient Numerical Computation&#8221;, Computing in Science &amp; Engineering, 13, 22-30 (2011)</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="id23" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id14">[8]</a></td><td>John D. Hunter. Matplotlib: A 2D Graphics Environment, Computing in Science &amp; Engineering, 9, 90-95 (2007)</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="id24" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id15">[9]</a></td><td>Jones E, Oliphant E, Peterson P, et al. &#8220;SciPy: Open Source Scientific Tools for Python&#8221;, 2001, <a class="reference external" href="http://www.scipy.org/">http://www.scipy.org/</a></td></tr>
</tbody>
</table>
</div>
</div>
<div class="section" id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><a class="reference internal" href="genindex.html"><em>Index</em></a></li>
<li><a class="reference internal" href="py-modindex.html"><em>Module Index</em></a></li>
<li><a class="reference internal" href="search.html"><em>Search Page</em></a></li>
</ul>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="align_traj_on_density.html" title="align_traj_on_density module"
             >next</a> |</li>
        <li><a href="#">Python modules for OpenStructure 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2015, Niklaus Johner.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2.3.
    </div>
  </body>
</html>